### ISSUE TITLE
EthStorage Devs Meeting #98 Agenda

### ISSUE THREAD

**Issue by qzhodl** [ethstorage/pm#130]
### Meeting Info
- Sep 24, 2024, 11:00 UTC+8
- Duration: 60 minutes
- [Meeting link](https://meeting.tencent.com/dm/DAnDVeZikxPP)
- Anyone can comment on the agenda where they would like this to be discussed.
### Agenda
- Protocol/App/Research progress
- BD/Marketing progress

**Comment by qzhodl** [comment]
Goal:
- Upstream L2 BLOB
- Audit storage contract

Updates:
- L2 BLOB PR opened: https://github.com/ethereum-optimism/specs/pull/383
- Reviewed some PRs from L2 and storage contract
- Study Holocene derivation [spec](https://github.com/ethereum-optimism/specs/pull/357/files)

TODO:
- Get feedbacks from OP
- Review storage contract

**Comment by blockchaindevsh** [comment]
Goal:
- Prepare WSC L2 testnet launch

Updates:

1. merged refactor PRs:
    1. https://github.com/ethstorage/optimism/pull/60
    2. https://github.com/ethstorage/optimism/pull/59
    3. https://github.com/ethstorage/optimism/pull/61
    4. https://github.com/ethstorage/op-geth/pull/5
2. created a PR merging upstream op-geth: https://github.com/ethstorage/op-geth/pull/8


TODO: 

1. merge upstream monorepo. 
2. prepare a testnet with fault-proof and technical docs

**Comment by Akagi201** [comment]
Goal:

Implement parallel EVM in op-geth

Updates:

[Implement conflict storage slot analysis](https://github.com/Akagi201/blockstm-go/commit/d0092bd2d6fa7350a94e4df0f69149777cdf14cd)

TODO:

Implement parallel block executor and test against real blocks.

**Comment by yazhenwish** [comment]
participated token2049 in Singapore

**Comment by iteyelmp** [comment]
1. https://github.com/ethstorage/ethstorage-sdk/pull/25
2. https://github.com/ethstorage/evm-large-storage/pull/11
3. https://github.com/ethstorage/ethfs-cli/pull/20
4. https://github.com/ethstorage/ethstorage-sdk/pull/26
5. Debugging gateway access file 404 problem.

**Comment by syntrust** [comment]
- https://github.com/ethstorage/es-node/pull/289
- Gateway certificate update for new chains
- Gateway 404
  - https://github.com/ethstorage/es-node/pull/318
  - https://github.com/ethstorage/es-op-geth/pull/1
- https://github.com/ethstorage/storage-contracts-v1/pull/110
- https://github.com/ethstorage/storage-contracts-v1/pull/111

**Comment by ping-ke** [comment]
- es-node integration: https://github.com/ethstorage/es-node/issues/308
  - [x] Code change:  
    - [ ] ES-Node Changes: https://github.com/ethstorage/es-node/pull/313 (Under review)

- EthStorage x Taiko: 
  - [x] investigation: read docs and go through the taiko codes 
  - [ ] run test node and client (under test and debug) 
  - [ ] change contract code (under test and debug) 
  - [ ] integration test (blocked by less of SGX)

**Comment by GrapeBaBa** [comment]
Goal: 
- Refine soul gas token design/spec

Updates:
- Read origin [design](https://perfect-amphibian-929.notion.site/Introducing-World-Supercomputer-L2-Paving-the-Way-to-the-Future-Web3-b18d1c2a605c42609b421dc962ac402e?pvs=4) from Qi Zhou, and try to understand.
- Start a spec PR for this.
- Learn Optimism MT-Cannon, it introduce a new syscall `futex` support synchronization for threads.

Todo:
- Continue work on the spec PR

**Comment by qizhou** [comment]
Goal:
- Prepare WSC L2 testnet/website/roadmap launch

Updates:
- Reviewed/Discussed Alt-DA spec by Qiang
- Reviewed some new PR for fault-proof support (https://perfect-amphibian-929.notion.site/OP-1042e493e88380f9ac96cf8e14d65fff?pvs=4)
- Shared our WSC roadmap with community members
- Discussed with Alan about preparing the technical docs
- Gave a sharing session about SoulGasTOken

TODO:
- With Alan, to prepare a testnet with fault-proof and technical docs
- Will prepare an OP Stack fee with a custom gas token

**Comment by dajuguan** [comment]
Goal:
E2E test of 4-ary dispute game

Updates:

- Implemented [calldata decoding test](https://github.com/ethstorage/optimism/commit/a73b45bb585dcd5caa783efce4fb19a2501ed547)
- Investigate how to mock EventLog
- Investigate how to communicate with OP
- Consult Alan and GrapeBaBa on how to engage with the community

TODO:

- Implement eventlog test for calldata decoding
- Write a simple proposal in OP-discord and get some feedback
- Study OP's core codes(channel/frame, fault proof, reorg) and write some blogs
